### PR TITLE
[#47] added event endpoint which transfers messages with at-least-once qos

### PR DESCRIPTION
**Pull request by dguggemos**
- added Artemis configuration image
 - start Artemis broker in example
 - extended Hono Client and example to allow event sender/receiver

### CODE REVIEW

File: config/qpid/qdrouterd.conf [hunk 1]
```diff
@@ -38,5 +38,5 @@ log {
      # One of [‘ROUTER’, ‘ROUTER_CORE’, ‘ROUTER_HELLO’, ‘ROUTER_LS’, ‘ROUTER_MA’, ‘MESSAGE’, ‘SERVER’, ‘AGENT’, ‘CONTAINER’, ‘CONFIG’, ‘ERROR’, ‘DISPATCH’, ‘POLICY’, ‘DEFAULT’]
     module: DEFAULT
      # Levels are: trace, debug, info, notice, warning, error, critical
-    enable: info+
+    enable: debug+
```
Comment (ppatierno) [inline config/qpid/qdrouterd.conf]: Why the debug level here ?

File: client/src/main/java/org/eclipse/hono/client/MessageSender.java [hunk 2]
```diff
@@ -8,22 +8,25 @@
...
 
 /**
  * A client for uploading telemetry data to a Hono server.
  *
  */
-public interface TelemetrySender {
+public interface MessageSender {
 
     /**
      * Checks if this sender can send or buffer (and send later) a telemetry message.
```
Comment (sophokles73) [inline client/src/main/java/org/eclipse/hono/client/MessageSender.java]: s /`telemetry message`/`message`?  

this applies to all JavaDoc in this interface ...

File: client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.proton.ProtonDelivery;
+import io.vertx.proton.ProtonHelper;
+
+/**
+ * A Vertx-Proton based client for uploading telemtry data to a Hono server.
```
Comment (sophokles73) [inline client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java]: this is no longer about `telemetry data` only, is it?

File: client/src/main/java/org/eclipse/hono/client/impl/EventConsumerImpl.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonReceiver;
+
+/**
+ * A Vertx-Proton based client for consuming telemetry data from a Hono server.
```
Comment (sophokles73) [inline client/src/main/java/org/eclipse/hono/client/impl/EventConsumerImpl.java]: for `consuming events` ...

File: client/src/main/java/org/eclipse/hono/client/impl/EventSenderImpl.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.proton.ProtonConnection;
+import io.vertx.proton.ProtonQoS;
+import io.vertx.proton.ProtonSender;
+
+/**
+ * A Vertx-Proton based client for event messages to a Hono server.
```
Comment (sophokles73) [inline client/src/main/java/org/eclipse/hono/client/impl/EventSenderImpl.java]: *publishing* `event messages`?

**Pull request merged**